### ISSUE TITLE
Creation of Super Admin Account fails / Unknown email configuration "debug"

### ISSUE THREAD

**Issue by jchristlieb** [foodcoopshop/foodcoopshop#546]
Hello, 

I am trying to set up the foodcoopshoop for a Berlin based food collective and have some issues with sending mails. See below a detailed description of what I did, the detailed error log and my mail config in credentials.php and custom_config.php. Thanks in advance for everything that helps me solve the issue. 

### Was hast du gemacht? What you did?

1. Installed FoodCoopShop v3.02. 
2. Registered a new user account at /sign-in route. 
3. Edited the record withing fcs_customer according to the user manual 
4. called /request-new-password route, entered the email for the superadmin account.

### Was ist passiert? What happened?
EXPLAIN WHAT IS ACTUALLY HAPPENING, HERE.

1. got redirect to an error page. 
2. checked the foodcoopshop/logs.error.log and received the following error message with stack trace

> 2020-07-24 10:41:13 Error: Fatal Error (1): Uncaught InvalidArgumentException: **Unknown email configuration "debug"**. in 
> /home/sites/site100019514/web/foodcoopshop/vendor/cakephp/cakephp/src/Mailer/Mailer.php:411
> Stack trace:
> #0 /home/sites/site100019514/web/foodcoopshop/src/Log/Engine/FileAndEmailLog.php(76): Cake\Mailer\Mailer->setProfile()
> #1 /home/sites/site100019514/web/foodcoopshop/src/Log/Engine/FileAndEmailLog.php(32): App\Log\Engine\FileAndEmailLog->sendEmailWithErrorInformation()
> #2 /home/sites/site100019514/web/foodcoopshop/vendor/cakephp/cakephp/src/Log/Log.php(390): App\Log\Engine\FileAndEmailLog->log()
> #3 /home/sites/site100019514/web/foodcoopshop/vendor/cakephp/cakephp/src/Log/Log.php(459): Cake\Log\Log::write()
> #4 /home/sites/site100019514/web/foodcoopshop/vendor/cakephp/cakephp/src/Error/ErrorLogger.php(80): Cake\Log\Log::error()
> #5 /home/sites/site100019514/web/foodcoopshop/vendor/cakephp/cakephp/src/Error/BaseErrorHandler.php(343): Cake\Error\ErrorLogger->log()
> #6 /home/sites/site100019514/web/foodcoopshop/vendor/cakephp/cakephp/src/Er in [/home/sites/site100019514/web/foodcoopshop/vendor/cakephp/cakephp/src/Mailer/Mailer.php, line 411]
> Request URL: /neues-passwort-anfordern
> Referer URL: https://foodkollektiv-wedding.de/neues-passwort-anfordern
> Client IP: 90.187.54.169
> Trace:
> Cake\Error\BaseErrorHandler::handleFatalError() - CORE/src/Error/BaseErrorHandler.php, line 258
> Cake\Error\BaseErrorHandler::Cake\Error\{closure}() - CORE/src/Error/BaseErrorHandler.php, line 122
> [main] - [internal], line ?? 

### Was sollte passieren? What you expected to happen?
EXPLAIN WHAT IS TO BE EXPECTED, HERE.

I expected the system to send an email to the stored mail address of the super user. This did not happen, thus believe my email configuration is not correct. How did I configure mailing?

**1. credentials.php configuration:**

> return [
> 
>     'app' => [
> 
>         /**
>          * the cronjobs for sending order lists and invoices need the credentials to a valid superadmin account
>          */
>         'adminEmail' => 'superadmin@foodkollektiv-wedding.de',
>         'adminPassword' => '************',
> 
>         /**
>          * set to true if you want to receive debug emails when exceptions are triggered
>          */
>         'emailErrorLoggingEnabled' => true,
> 
>         /**
>          * TO address
>          * when exceptions are triggered, emails are sent TO this email address
>          */
>         'debugEmail' => 'mail@example.com',
> 
>         /**
>          * this email address
>          * - receives the test email of url: /email /admin/configurations/sendTestEmail
>          * - must have a valid customer record (which is never shown in customer list)
>          * - receives the database dumps of BackupDatabase Shell
>          */
>         'hostingEmail' => 'superadmin@foodkollektiv-wedding.de',
> 
>         /**
>          * FALLBACK email config
>          * when the main email config in custom_config.php is wrong, you can define this fallback email config
>          * to send the emails despite a wrong main config
>          * if you don't want to use the email fallback, leave it commented
>          */
> //         'EmailTransport' => [
> //             'fallback' => [
> //                 'emailFormat' => 'html',
> //                 'className' => 'Smtp',
> //                 'host' => 'mail.manitu.de',
> //                 'port' => 465,
> //                 'timeout' => 30,
> //                 'username' => '*********@foodkollektiv-wedding.de',
> //                 'password' => '*************',
> //                 'client' => null,
> //                 'tls' => null,
> //             ],
> //         ],
> 
>     ],
> 
>     /**
>      * DEBUG email config
>      * emails are sent FROM this email configuration if emailErrorLoggingEnabled is set to true
>      * needs to be configured to run BackupDatabaseShell
>      */
> //     'EmailTransport' => [
> //         'debug' => [
> //             'emailFormat' => 'html',
> //             'className' => 'Smtp',
> //             'host' => 'mail.manitu.de',
> //             'port' => 465,
> //             'timeout' => 30,
> //             'username' => '**********@foodkollektiv-wedding.de',
> //             'password' => '***************',
> //             'client' => null,
> //             'tls' => null,
> //         ],
> //     ],
> //     'Email' => [
> //         'debug' => [
> //             'from' => ['example@example.com' => 'Example'],
> //             'charset' => 'utf-8',
> //             'headerCharset' => 'utf-8',
> //         ],
> //     ],
> 
> ];

**2. custom_config.php configuration:** 

> return [
>     'debug' => false,
>     'EmailTransport' => [
>         'default' => [
>             'className' => 'Smtp',
>             'host' => 'mail.manitu.de',
>             'port' => 465,
>             'timeout' => 30,
>             'username' => '*******@foodkollektiv-wedding.de',
>             'password' => '************',
>             'client' => null,
>             'tls' => true,
>         ]
>     ],
>     'Email' => [
>         'default' => [
>             'from' => [],
>         ]
>     ],
>     'Datasources' => [
>         'default' => [
>             'host' => '********',
>             'username' => '*********',
>             'password' => '*********',
>             'database' => '*********',
>             'prefix' => 'fcs_'
>         ]
>     ],
> 
>     /**
>      * A random string used in security hashing methods.
>      */
>     'Security' => [
>         'salt' => '****************************************'
>     ],
> 
>     'app' => [
>         
>         'discourseSsoEnabled' => false,
> 
>         /**
>          * A random string used for Discourse SSO
>          */
>         'discourseSsoSecret' => '',
> 
>         /**
>          * optional: message that is displayed in the dialog where order-detail status can be changed (/admin/order-details)
>          */
>         'additionalOrderStatusChangeInfo' => '',
> 
>         /**
>          * your host's name, eg. http://www.yourfoodcoop.com
>          */
>         'cakeServerName' => 'https://foodkollektiv-wedding.de',
> 
>         /**
>          * whether to apply a member fee to the members account balance
>          */
>         'memberFeeEnabled' => false,
> 
>         /**
>          * cronjob needs to be activated / deactivated too if you change emailOrderReminderEnabled
>          * @see https://foodcoopshop.github.io/en/cronjobs
>          */
>         'emailOrderReminderEnabled' => true,
> 
>         /**
>          * valid options of array: 'cashless' or 'cash' (or both but this is not recommended)
>          */
>         'paymentMethods' => [
>             'cashless'
>         ]
> 
>     ]
> ];

**Comment by mrothauer** [comment]
Hi, your debug email configuration needs to be set up properly. See https://github.com/foodcoopshop/foodcoopshop/issues/369

**Comment by jchristlieb** [comment]
Hi @mrothauer, 

thanks for pointing me in the right direction. It seems that the initial email address I tried to set up as a mail driver did not work. Changed to a GMAIL address for testing purpose and everything worked well. 

Moreover during the installation I was wondering why the EmailTransport example configuration is commented out by default?  For me it wasn't clear from the documentation that the comment need to be removed (even though technically it makes total sense).
